Distribution/Server/Users/AuthToken.hs

```diff
@@ -0,0 +1,47 @@
+{-# LANGUAGE DeriveDataTypeable, GeneralizedNewtypeDeriving, TemplateHaskell #-}
+{-# LANGUAGE TypeFamilies #-}
+module Distribution.Server.Users.AuthToken
+    ( AuthToken(..)
+    , generateAuthToken
+    )
+where
+
+import Distribution.Server.Framework.MemSize
+
+import System.Random.MWC
+import Distribution.Text
+         ( Text(..) )
+import qualified Distribution.Compat.ReadP as Parse
+import qualified Text.PrettyPrint          as Disp
+import qualified Data.Char as Char
+import qualified Data.Text as T
+import qualified Data.Text.Encoding as T
+import qualified Data.Vector as V
+import qualified Data.ByteString as BS
+import qualified Data.ByteString.Base16 as BS16
+
+import Control.Applicative ((<$>))
+import Data.Aeson (ToJSON, FromJSON)
+import Data.SafeCopy (base, deriveSafeCopy)
+import Data.Typeable (Typeable)
+
+newtype AuthToken = AuthToken T.Text
```

Comment:
Does this need to be Text? It's just base64 encoded bytes right?

I'd suggest we internally represent the token as the raw bytes (in a ShortByteString) and only convert to/from the base64 when rendering (e.g. in the ToJSON instance).